api/src/main/java/au/org/aodn/nrmn/restapi/validation/process/ValidationProcess.java

```diff
@@ -627,8 +626,34 @@ public ValidationError validateSurveyGroup(String surveyKey, List<StagedRowForma
         return rows.stream().map(stagedRow -> mapper.map(stagedRow, StagedRowFormatted.class)).collect(Collectors.toList());
     }
 
-    public ValidationResponse process(StagedJob job) {
+    public ValidationResponse generateSummary(Collection<StagedRowFormatted> mappedRows) {
         ValidationResponse response = new ValidationResponse();
+        
+        Collection<String> distinctSites = mappedRows.stream().map(r ->r.getRef().getSiteCode().trim()).filter(s -> s!= null).distinct().collect(Collectors.toList());
+        Collection<String> distinctSitesExisting = mappedRows.stream().map(r ->r.getSite().getSiteCode()).filter(s -> s != null).distinct().collect(Collectors.toList());
```

Comment:
I think the filter should come first here, so:
```java
Collection<String> distinctSites = mappedRows.stream().filter(s -> s.getRef() != null).map(r -> r.getRef().getSiteCode().trim()).distinct().collect(Collectors.toList());
Collection<String> distinctSitesExisting = mappedRows.stream().filter(s -> s.getSite() != null).map(r -> r.getSite().getSiteCode()).distinct().collect(Collectors.toList());
```

Otherwise if `r.getRef()` or more likely `r.getSite()` yield null then the infamous NullPointerException will occur.

Reply:
ref will never be null but site could be. Fixed